smartsnippets/models.py

```diff
@@ -269,5 +271,9 @@ def remove_cached_variables(instance, **kwargs):
     if key in cache:
         cache.delete(key)
 
+def clean_variable_name(variable_name):
+    """ Convert the variable name to a name that can be used as a template variable. """
+    return re.sub(r"[^a-zA-Z0-9_]", '', variable_name.replace(' ', '_'))
```

Comment:
Why perform substitutions under the hood rather than just flagging this as an error and let the user correct it for himself?